templates/ui-frameworks/vue/collection/ui/src/{{dna_role_name}}/{{coordinator_zome_manifest.name}}/{{pascal_case collection_name}}.vue.hbs

```diff
@@ -15,73 +15,53 @@
   </div>
 </template>
 
-<script lang="ts">
-import { inject, toRaw, ComputedRef, defineComponent } from 'vue';
-import { decode } from '@msgpack/msgpack';
-import { AppClient, NewEntryAction, Link, Record, AgentPubKey, EntryHash, ActionHash, HolochainError } from '@holochain/client';
+<script setup lang="ts">
+import { ref, onMounted, toRaw, inject{{#if (eq collection_type.type "ByAuthor")}}, defineProps{{/if}} } from 'vue';
+import { NewEntryAction, Link, AgentPubKey, ActionHash, HolochainError } from '@holochain/client';
```

Comment:
Unused imports
```suggestion
import { Link, ActionHash, HolochainError } from '@holochain/client';
```

Reply:
NewEntryAction is used under some conditions, would be nice to only import it under those conditions though. AgentPubKey is used on line 30

Reply:
> Unused imports

Intentional, because some scaffold commands will generate ui code that will use those imports while others don't. I have explored some ways to address this in the past but it ended up adding more complexity than needed, I can open a spike issue to look into this separately though.

Reply:
Removed `NewEntryAction`.

`AgentPubKey` is still relevant to this file, in the case where a by-author collection is scaffolded

Reply:
Wait but we are still conditionally using `NewEntryAction` on line 64

Reply:
Oh yes, we still do, good catch @mattyg. All imports are "used" in this file.